backend/src/models/entry/entryAnalysis.ts

```diff
@@ -85,64 +73,36 @@ async function removeLegacyApiKey(configId: string): Promise<string> {
       }
     }
   }
-  return config.model.analysis;
-}
 
-async function getAnalysisCompletion(
-  configModelAnalysis: string,
-  content: string
-): Promise<any> {
-  if (process.env.OPENAI_API_KEY === undefined) {
-    throw new Error('OPENAI_API_KEY undefined. Must be set to get analysis');
+  if (!process.env.OPENAI_API_KEY) {
+    throw new Error('API key is not available)');
   }
-  const cdGpt = new CdGpt(
-    process.env.OPENAI_API_KEY,
-    configModelAnalysis,
-    '',
-    0.7
-  );
+
+  const cdGpt = new CdGpt(process.env.OPENAI_API_KEY, config.model.analysis);
 
   cdGpt.seedAnalysisMessages();
   cdGpt.addUserMessage({ analysis: content });
 
-  const analysisCompletion: any = await cdGpt.getAnalysisCompletion();
+  const analysisCompletion = await cdGpt.getAnalysisCompletion();
 
   if (analysisCompletion.error) {
     throw new Error(analysisCompletion.error.message);
   }
 
-  return JSON.parse(analysisCompletion.choices[0].message.content);
-}
+  const response = JSON.parse(analysisCompletion.choices[0].message.content);
 
-// Get the analysis content for an entry
-entryAnalysisSchema.methods.getAnalysisContent = async function (
-  configId: string,
-  content: string
-): Promise<any> {
-  const configModelAnalysis = await removeLegacyApiKey(configId);
-  const response = await getAnalysisCompletion(configModelAnalysis, content);
-  const {
-    reframed_thought: reframing,
-    distortion_analysis: analysis,
-    impact_assessment: impact,
-    affirmation,
-    is_healthy: isHealthy,
-  } = response;
+  const { reframed_thought: reframing, distortion_analysis: analysis, impact_assessment: impact, affirmation, is_healthy: isHealthy } = response;
```

Comment:
This line is a big reason why I decided to leave the type as any for now. There's a lot of things getting unpacked from a string, and it's extremely similar to `FormatInstructions`, so it seems like there's a missing type here for `Message.content`

Reply:
I see what you're saying. This is even more of a reason why this should be pulled out into CdGpt. It seems like it belongs in CdGpt even more now with `FormatInstructions` defined. I'm not sure though if we can define a type for this because the parsed response above this line is coming from an external source.